tests/acceptance/comparative_booster_study/variables_lib.py

```diff
@@ -0,0 +1,140 @@
+import operator
+from functools import reduce
+
+from databuilder.codes import CTV3Code, ICD10Code
+from databuilder.query_language import case, when
+
+from . import schema
+
+
+def create_sequential_variables(
+    dataset, variable_name_template, events, column, num_variables, sort_column=None
+):
+    sort_column = sort_column or column
+    for index in range(num_variables):
+        next_event = events.sort_by(getattr(events, sort_column)).first_for_patient()
+        events = events.take(
+            getattr(events, sort_column) > getattr(next_event, sort_column)
+        )
+        variable_name = variable_name_template.format(n=index + 1)
+        setattr(dataset, variable_name, getattr(next_event, column))
+
+
+def _registrations_overlapping_period(start_date, end_date):
+    regs = schema.practice_registrations
+    return regs.take(
+        regs.start_date.is_on_or_before(start_date)
+        & (regs.end_date.is_after(end_date) | regs.end_date.is_null())
+    )
+
+
```

Comment:
The functions below here are a pretty good practial proof-of-concept for reusable variables. Seems like it'll work out pretty well?